From 8c951e9300b48b0f86073e9ff5642469dd50a137 Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Sun, 5 Mar 2017 14:51:54 +0200
Subject: [PATCH 0845/1345] dts: cp110: fix eip 197 nodes

commit  c0c2a8b6fc659bb3f3b4290e128aa2c8440fed64 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch alignes the eip197 nodes with the new
eip 197 driver.

Change-Id: I6eeaafab3435cfab727bfe4d1940fece5172f9f0
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37214
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi |   19 ++++++++++---------
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi   |   19 ++++++++++---------
 2 files changed, 20 insertions(+), 18 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index 79529d2..6f3ddfc 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -352,16 +352,17 @@ cps_trng: trng@760000 {
 };
 
 cps_eip197: eip197@800000 {
-	compatible = "marvell,eip197";
-	reg = <0x800000 0x100000 0x880000 0x1000 0x881000 0x1000 0x882000 0x1000 0x883000 0x1000>;
-	reg-names = "global", "ring0", "ring1", "ring2", "ring3";
+	compatible = "inside-secure,safexcel-eip197";
+	reg = <0x800000 0x200000>;
 	dma-coherent;
-	interrupts = <ICU_GRP_NSR 88 IRQ_TYPE_LEVEL_HIGH>,
-			<ICU_GRP_NSR 89 IRQ_TYPE_LEVEL_HIGH>,
-			<ICU_GRP_NSR 90 IRQ_TYPE_LEVEL_HIGH>,
-			<ICU_GRP_NSR 91 IRQ_TYPE_LEVEL_HIGH>,
-			<ICU_GRP_NSR 92 IRQ_TYPE_LEVEL_HIGH>;
-	interrupt-names = "ring0", "ring1", "ring2", "ring3", "global";
+	interrupts = <ICU_GRP_REI 87 IRQ_TYPE_LEVEL_HIGH>,
+		     <ICU_GRP_NSR 88 IRQ_TYPE_LEVEL_HIGH>,
+		     <ICU_GRP_NSR 89 IRQ_TYPE_LEVEL_HIGH>,
+		     <ICU_GRP_NSR 90 IRQ_TYPE_LEVEL_HIGH>,
+		     <ICU_GRP_NSR 91 IRQ_TYPE_LEVEL_HIGH>,
+		     <ICU_GRP_NSR 92 IRQ_TYPE_LEVEL_HIGH>;
+	interrupt-names = "eip_addr", "ring0", "ring1",
+			  "ring2", "ring3", "eip_out";
 	clocks = <&cps_syscon0 1 26>;
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 868904c..599b725 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -450,16 +450,17 @@ cpm_trng: trng@760000 {
 };
 
 cpm_eip197: eip197@800000 {
-	compatible = "marvell,eip197";
-	reg = <0x800000 0x100000 0x880000 0x1000 0x881000 0x1000 0x882000 0x1000 0x883000 0x1000>;
-	reg-names = "global", "ring0", "ring1", "ring2", "ring3";
+	compatible = "inside-secure,safexcel-eip197";
+	reg = <0x800000 0x200000>;
 	dma-coherent;
-	interrupts = <ICU_GRP_NSR 88 IRQ_TYPE_LEVEL_HIGH>,
-			<ICU_GRP_NSR 89 IRQ_TYPE_LEVEL_HIGH>,
-			<ICU_GRP_NSR 90 IRQ_TYPE_LEVEL_HIGH>,
-			<ICU_GRP_NSR 91 IRQ_TYPE_LEVEL_HIGH>,
-			<ICU_GRP_NSR 92 IRQ_TYPE_LEVEL_HIGH>;
-	interrupt-names = "ring0", "ring1", "ring2", "ring3", "global";
+	interrupts = <ICU_GRP_REI 87 IRQ_TYPE_LEVEL_HIGH>,
+		     <ICU_GRP_NSR 88 IRQ_TYPE_LEVEL_HIGH>,
+		     <ICU_GRP_NSR 89 IRQ_TYPE_LEVEL_HIGH>,
+		     <ICU_GRP_NSR 90 IRQ_TYPE_LEVEL_HIGH>,
+		     <ICU_GRP_NSR 91 IRQ_TYPE_LEVEL_HIGH>,
+		     <ICU_GRP_NSR 92 IRQ_TYPE_LEVEL_HIGH>;
+	interrupt-names = "eip_addr", "ring0", "ring1",
+			  "ring2", "ring3", "eip_out";
 	clocks = <&cpm_syscon0 1 26>;
 	status = "disabled";
 };
-- 
1.7.9.5

